cmd/zetaclientd/start.go

```diff
@@ -115,11 +110,22 @@ func Start(_ *cobra.Command, _ []string) error {
 		graceful.ShutdownNow()
 	})
 
-	maintenance.NewShutdownListener(zetacoreClient, logger.Std).Listen(ctx, func() {
+	shutdownListener := maintenance.NewShutdownListener(zetacoreClient, logger.Std)
+	if err := shutdownListener.RunPreStartCheck(ctx); err != nil {
+		return errors.Wrap(err, "pre start check failed")
+	}
+	shutdownListener.Listen(ctx, func() {
 		logger.Std.Info().Msg("Shutdown listener received an action to shutdown zetaclientd.")
 		graceful.ShutdownNow()
 	})
 
+	// This will start p2p communication so it should only happen after
+	// preflight checks have completed
+	tss, err := zetatss.Setup(ctx, tssSetupProps, logger.Std)
```

Comment:
To keep the code cohesive, let's move all TSS-related stuff AFTER `shutdownListener`.

```suggestion
granteePubKeyBech32, err := resolveObserverPubKeyBech32(cfg, passes.hotkey)
	if err != nil {
		return errors.Wrap(err, "unable to resolve observer pub key bech32")
	}

	tssSetupProps := zetatss.SetupProps{
		Config:              cfg,
		Zetacore:            zetacoreClient,
		GranteePubKeyBech32: granteePubKeyBech32,
		HotKeyPassword:      passes.hotkey,
		TSSKeyPassword:      passes.tss,
		BitcoinChainIDs:     btcChainIDsFromContext(appContext),
		PostBlame:           isEnvFlagEnabled(envFlagPostBlame),
		Telemetry:           telemetry,
	}

	tss, err := zetatss.Setup(ctx, tssSetupProps, logger.Std)
	if err != nil {
		return errors.Wrap(err, "unable to setup TSS service")
	}

	isObserver, err := isObserverNode(ctx, zetacoreClient)
	switch {
	case err != nil:
		return errors.Wrap(err, "unable to check if observer node")
	case !isObserver:
		logger.Std.Warn().Msg("This node is not an observer node. Exit 0")
		return nil
	}
```

Reply:
Also, we need to ensure the following sequence:
1. `shutdownListener.RunPreStartCheck()`
2. `tss.Setup()`
3.  `tssListener.Listen()` 🚨
4. `shutdownListener.Listen()`

Otherwise when `tss.Setup()` **might** init a new key sign ceremony the tssListener will kill the client before it finishes the procedure.

Reply:
Ok I updated this.